### PR TITLE
[codesign] white list for debug tools

### PR DESCRIPTION
**Pull request by XilaiZhang**
skip codesign for engine binaries that are debug symbols and not executable, which don't need to be codesigned or notarized.

Context: https://github.com/flutter/flutter/issues/124322

### CODE REVIEW

File: codesign/lib/src/file_codesign_visitor.dart [hunk 1]
```diff
@@ -71,6 +71,7 @@ class FileCodesignVisitor {
     'CODESIGN_TEAM_ID': '',
     'APP_SPECIFIC_PASSWORD': ''
   };
+  Set<String> whitelist = {'FlutterEmbedder.framework.zip', 'FlutterMacOS.dSYM.zip'};
```
Comment (godofredoc) [inline codesign/lib/src/file_codesign_visitor.dart:74]: I would prefer to have something closer to the artifact creation. Doing this way will mean that we need to update this tool every time a new artifact is added.

Comment (godofredoc) [reply codesign/lib/src/file_codesign_visitor.dart:74]: Maybe just logging that a file was not code signed because it didn't have code sign metadata?

Comment (godofredoc) [reply codesign/lib/src/file_codesign_visitor.dart:74]: @christopherfujino what do you think?

Comment (XilaiZhang) [reply codesign/lib/src/file_codesign_visitor.dart:74]: Ah i see. Good point! yeah need to change BUILD.gn to add another file

Comment (christopherfujino) [reply codesign/lib/src/file_codesign_visitor.dart:74]: Yeah, I think the build rules is the right place to add this.

File: codesign/lib/src/file_codesign_visitor.dart [hunk 2]
```diff
@@ -64,6 +64,7 @@ class FileCodesignVisitor {
 
   Set<String> fileWithEntitlements = <String>{};
   Set<String> fileWithoutEntitlements = <String>{};
+  Set<String> fileWhitelist = <String>{};
```
[the remark this answers is not in the document]
Comment (XilaiZhang) [reply codesign/lib/src/file_codesign_visitor.dart:67]: umm when I googled the term whitelist it says its a computing word for "a list of people or things considered to be acceptable or trustworthy"

Anyways we will be abandoning this PR in favor of the namelist approach of zip files in recipe

**Comment by XilaiZhang** [comment]
We will use namelist approach of zip files in recipe implemented by sir Godofredo

**Pull request closed without merging**